src/iterable.cr

```diff
@@ -33,4 +33,39 @@ module Iterable
   def each_with_object(obj)
     each.with_object(obj)
   end
+
+  # :nodoc:
+  # overrides Comparable.==
+  def ==(other: self)
+# TODO: if T is a noncontainer value && Enumerable is allocated as an array use memcmp()
+    equals?(other) { |x, y| x == y }
+  end
+
+  # Iterates over both Enumerables comparing each value against other.value.
```

Comment:
There's no `other.value` method (called). Let's try to stay with explanatory wording here, not pseudo code.